README.md

```diff
@@ -1,2 +1,57 @@
-# example-application
-Example out-of-tree application that is also a module
+# Zephyr Example Application
+
+This repository contains a Zephyr example application. The main purpose of this
+repository is to serve as a reference on how to structure Zephyr based
+applications. Some of the features demonstrated in this example are:
+
+- Basic application skeleton
+- Custom boards
+- Custom Devicetree bindings
+- Out-of-tree drivers
+- Documentation using Doxygen and Sphinx
+- Example CI configuration (using Github Actions)
+
+## Getting Started
+
+Before getting started, make sure you have a proper Zephyr development
+environment. You can follow the official
+[Zephyr Getting Started Guide](https://docs.zephyrproject.org/latest/getting_started/index.html).
+
+### Initialization
+
+First of all create a folder where you want to setup the application workspace.
+Then, from the workspace folder initialize the application by running:
+
+```shell
+west init -m git@github.com:zephyrproject-rtos/example-application --mr main
+```
+
+After that, make sure all modules are up to date by running, from the
+just initialized `example-application` folder:
+
+```shell
+west update
+```
+
+This last step should be run every time you want to synchronize with Zephyr.
```

Comment:
I think locking the manifest to v2.5.0, and updating it when we cut new releases, might be better. I guess most people won't want to live on the bleeding edge.